### PR TITLE
Select

### PR DESCRIPTION
**Pull request by wlatanowicz**

### CODE REVIEW

File: src/lib/Select.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    right: 16px;
+    top: 50%;
+    width: 0;
+    pointer-events: none;
+  }
+`;
+
+const StyledSelect = styled.select`
+  -webkit-appearance: none;
+  -moz-appearance: none;
```
Comment (feimosi) [inline src/lib/Select.js]: Autoprefixer doesn't handle that?

File: src/lib/Select.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import PropTypes from 'prop-types';
+
+import { palette, style } from '../constants';
+
+const SelectWrapper = styled.div`
+  position: relative;
+  &:after {
+    border-color: ${palette.GRAY.black} transparent;
+    border-style: solid;
+    border-width: 6px 6px 0 6px;
```
Comment (feimosi) [inline src/lib/Select.js]: Would `border-width: 6px 6px 0;` work as well ?

File: src/lib/Select.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+);
+
+Select.defaultProps = {
+  valid: true,
+};
+
+Select.propTypes = {
+  placeholder: PropTypes.string,
+  children: PropTypes.arrayOf(PropTypes.object),
+};
```
Comment (feimosi) [inline src/lib/Select.js]: `options` and `valid` are missing here.

Btw. if you have some spare time, could you add ESLint to the project? That would catch many such things.

File: src/lib/Select.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+`;
+
+export const Option = styled.option`
+  color: ${palette.GRAY.black};
+`;
+
+
+const Select = (props) => (
+  <SelectWrapper>
+    <StyledSelect {...props} required={!!props.placeholder}>
```
Comment (feimosi) [inline src/lib/Select.js]: You should pass only the props that are not used in `Select`, like
`const { placeholder, ...rest } = props`
`<StyledSelect {...rest}`

And how's `placeholder` connected to select being `required` ?

Comment (wlatanowicz) [reply src/lib/Select.js]: AFAIK this is the only non-JS way to change color of text for "placeholder".
https://stackoverflow.com/a/8442831

Comment (wlatanowicz) [reply src/lib/Select.js]: Do you prefer the JS way?

File: src/lib/Select.js (hunk 1 above)
Comment (wlatanowicz) [reply src/lib/Select.js]: Sure it does :-)

File: src/lib/Select.js (hunk 4 above)
Comment (feimosi) [reply src/lib/Select.js]: Alright, I didn't know about this trick with `required`. I'd put a short comment about that though so other developers won't feel confused too.

Comment (wlatanowicz) [reply src/lib/Select.js]: OK

**Pull request merged**